packages/openapi-ts/src/utils/write/services.ts

```diff
@@ -616,9 +655,14 @@ export const processServices = async ({
 
   // Import all models required by the services.
   if (files.types && !files.types.isEmpty()) {
-    const importedTypes = imports
-      .filter(unique)
-      .map((name) => ({ asType: true, name }));
+    const importedTypes = imports.filter(unique).map((name) => {
+      const hasTransformer = client.types[name]?.hasTransformer === true;
+
+      return {
+        asType: !hasTransformer,
+        name,
+      };
+    });
```

Comment:
When a transformer function is paired with the type we don't want typescript to be using the `import { type MyType }` syntax, but this code was nicely written so easy to vary that on a per-type basis